pkg/ingress/kube/ingressv1/controller.go

```diff
@@ -507,47 +506,44 @@ func (c *controller) ConvertHTTPRoute(convertOptions *common.ConvertOptions, wra
 		}
 
 		wrapperHttpRoutes := make([]*common.WrapperHTTPRoute, 0, len(rule.HTTP.Paths))
+
 		for _, httpPath := range rule.HTTP.Paths {
+			var pathType common.PathType
+			if wrapper.AnnotationsConfig.NeedRegexMatch() {
+				pathType = common.Regex
+			} else {
+				switch *httpPath.PathType {
+				case ingress.PathTypeExact:
+					pathType = common.Exact
+				case ingress.PathTypePrefix:
+					pathType = common.Prefix
+				}
+			}
+
 			wrapperHttpRoute := &common.WrapperHTTPRoute{
 				HTTPRoute:     &networking.HTTPRoute{},
 				WrapperConfig: wrapper,
 				Host:          rule.Host,
 				ClusterId:     c.options.ClusterId,
 			}
-			httpMatch := &networking.HTTPMatchRequest{}
 
-			path := httpPath.Path
-			if wrapper.AnnotationsConfig.NeedRegexMatch() {
-				wrapperHttpRoute.OriginPathType = common.Regex
-				httpMatch.Uri = &networking.StringMatch{
-					MatchType: &networking.StringMatch_Regex{Regex: httpPath.Path + ".*"},
-				}
-			} else {
-				switch *httpPath.PathType {
-				case ingress.PathTypeExact:
+			// if the pathType is Prefix, it will have 2 httpMatches, the first is the type of Exact and the next is Prefix
+			// so if the pathType is Prefix, the wrapperHttpRoute.OriginPathType is still Prefix, though it has been assigned twice
+			httpMatches := c.generateHttpMatches(pathType, httpPath.Path, wrapperVS)
+			for _, httpMatch := range httpMatches {
```

Comment:
Here no need to traverse the httpMatches. We treat these two (at most) routes as one route for RouteCache to show for console.

The users of console still get the only one route, just the prefix match.

Reply:
I'm not sure if I misunderstood your point🤔, but my understanding is to not traverse through httpMatches, but to only keep the prefix match. 

But for example, if it's `prefix=/abc` or `prefix=/abc/` in Ingress, and we only keep the prefix match, then in the `wrapperHttpRoute.HTTPRoute.Match`, there will only be `prefix=/abc/`. In the current Higress version, this maybe not match the request `path=/abc`.